sunpy/visualization/animator/base.py

```diff
@@ -522,55 +500,112 @@ def _sanitize_axis_ranges(self, axis_ranges, data):
             else:
-                raise ValueError("axis_ranges must be None or a list with length equal to number "
-                                 "of axes in data whose elements are either None, [min,max], "
-                                 "or a list/array of same length as the plot/image axis of data.")
+                # Depending on length of axis_ranges[i], leave unchanged,
+                # convert to pixel centers or raise an error due to incompatible format.
+                axis_ranges[i] = np.asarray(axis_ranges[i])
+                if len(axis_ranges[i]) == 2:
+                    # Set extent.
+                    extent = extent + [axis_ranges[i][0], axis_ranges[i][-1]]
+                elif axis_ranges[i].ndim == 1 and len(axis_ranges[i]) == data_shape[i]+1:
+                    # If array of individual pixel edges supplied, first set extent
+                    # from first and last pixel edge, then convert axis_ranges to pixel centers.
+                    # The reason that pixel edges are required as input rather than centers
+                    # is so that the plot extent can be derived from axis_ranges (above)
+                    # and APIs using both [min, max] pair and manual definition of each pixel
+                    # values can be unambiguously and simultanously supported.
+                    extent = extent + [axis_ranges[i][0], axis_ranges[i][-1]]
+                    axis_ranges[i] = edges_to_centers_nd(axis_ranges[i])
+                elif axis_ranges[i].ndim == ndim and axis_ranges[i].shape[i] == data_shape[i]+1:
+                    extent = extent + [axis_ranges[i].min(), axis_ranges[i].max()]
+                    axis_ranges[i] = edges_to_centers_nd(axis_ranges[i], i)
+                else:
+                    raise ValueError(incompatible_axis_ranges_error_message(i))
+        # For each axis validate and translate the axis_ranges.
+        for i in self.slider_axes:
+            if axis_ranges[i] is None:
+                    axis_ranges[i] = np.arange(data_shape[i])
+            elif len(axis_ranges[i]) == 2:
+                axis_ranges[i] = np.linspace(axis_ranges[i][0], axis_ranges[i][-1],
+                                             data_shape[i]+1)
+                axis_ranges[i] = edges_to_centers_nd(axis_ranges[i])
+            elif len(axis_ranges[i]) == data_shape[i]+1:
+                # If array of individual pixel edges supplied, convert to pixel centers.
+                axis_ranges[i] = edges_to_centers_nd(np.asarray(axis_ranges[i]), i)
+            else:
+                raise ValueError(incompatible_axis_ranges_error_message(i))
 
-        return axis_ranges
+        return axis_ranges, extent
 
     @abc.abstractmethod
     def plot_start_image(self):  # pragma: no cover
         """
         Abstract method for plotting first slice of array.
 
-        Must exists here but be defined in subclass.
+        Must exist here but be defined in subclass.
 
         """
 
```

Comment:
Do we not want this to raise a `NotImplementedError`?

Reply:
That's a good idea.